### PR TITLE
Feature slugify

### PR DESCRIPTION
**Pull request by akazorg**
Please read [#557](https://github.com/the-control-group/voyager/pull/557). The other branch was accidentally deleted.

### CODE REVIEW

File: publishable/views/posts/edit-add.blade.php [hunk 1]
```diff
@@ -121,7 +121,10 @@
                         <div class="panel-body">
                             <div class="form-group">
                                 <label for="name">URL slug</label>
-                                <input type="text" class="form-control" name="slug" placeholder="slug" value="@if(isset($dataTypeContent->slug)){{ $dataTypeContent->slug }}@endif">
+                                <input type="text" class="form-control" id="slug" name="slug"
+                                    placeholder="slug"
+                                    @if(isset($dataTypeContent)){!! isFieldSlugAutoGenerator($dataTypeContent, "slug") !!}@endif
```
Comment (akazorg) [inline publishable/views/posts/edit-add.blade.php]: Added this verification, but Travis is still failing.

Comment (akazorg) [inline publishable/views/posts/edit-add.blade.php:132]: Added this verification, but Travis is still failing.

File: src/Http/Controllers/VoyagerBreadController.php [hunk 2]
```diff
@@ -168,13 +168,17 @@ public function create(Request $request)
         // Check permission
         Voyager::can('add_'.$dataType->name);
 
+        $dataTypeContent = (strlen($dataType->model_name) != 0)
+                            ? new $dataType->model_name()
+                            : false;
+
```
Comment (akazorg) [inline src/Http/Controllers/VoyagerBreadController.php:174]: Added ```$dataTypeContent```, but Travis is still failing.

**Review by akazorg — COMMENTED** [review]
I've tested adding and editing a Post and it works.

File: src/Helpers/helperSlugify.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+     * Determine the details field, for a given dataTypeContent.
+     *
+     * @param Illuminate\Database\Eloquent\Collection $dataTypeContent
+     * @param string                                  $field
+     *
+     * @return string HTML output.
+     */
+    function isFieldSlugAutoGenerator($dataTypeContent, $field)
+    {
+        $_tmp = (isset($dataTypeContent->id))
```
Comment (fergthh) [inline src/Helpers/helperSlugify.php:37]: If you add the following

``` php

if (! isset($dataType)) {
            return;
}

```

TravisCI should be pass

Comment (fergthh) [reply src/Helpers/helperSlugify.php:37]: travis error

Comment (akazorg) [reply src/Helpers/helperSlugify.php:37]: You're right! Thank you @fergthh, it passed.

Comment (akazorg) [reply src/Helpers/helperSlugify.php:37]: @fergthh, where can I find that log?

**Comment by akazorg** [comment]
Regarding documentation, user may configure the Slugify by editing the BREAD and placing:
```
{
    "slugify": {
        "origin": "title", 
        "forceUpdate": true    // Optional, default is false
    }
}
```

File: src/Helpers/helperSlugify.php (hunk 3 above)
Comment (fergthh) [reply src/Helpers/helperSlugify.php:37]: I added a break in the loop on the RouteTest file and I dumped the response. Something like this...

``` php

if($response->status() == 500)
{
      $this->dump();
      break;
}

```

Comment (akazorg) [reply src/Helpers/helperSlugify.php:37]: Thanks @fergthh. Next time I'll handle it that way.

File: publishable/database/seeds/DataRowsTableSeeder.php [hunk 4]
```diff
@@ -166,7 +166,7 @@ public function run()
                 'edit'         => 1,
                 'add'          => 1,
                 'delete'       => 1,
-                'details'      => '',
+                'details'      => '{"slugify": {"origin": "title", "forceUpdate": true}}',
```
Comment (marktopper) [inline publishable/database/seeds/DataRowsTableSeeder.php]: Can you update it to be ```
json_encode([
    'slugify' => [
        'origin' => 'title',
        'forceUpdate' => true,
    ],
]),
```

Might be a little cleaner, I will cleanup the other rows.

File: src/Helpers/helperSlugify.php (hunk 3 above)
Comment (marktopper) [reply src/Helpers/helperSlugify.php:37]: I pushed a change into your other pull request that handled exceptions.

Add `$this->disableExceptionHandling();` to the start of `RouteTest@testGetRoutes`.
That change should be committed into `master` for better understanding of the route failures.

Comment (marktopper) [reply src/Helpers/helperSlugify.php:37]: I will push a commit for this.

**Comment by marktopper** [comment]
Create work, will test it out before I merge.

**Comment by akazorg** [comment]
We could expose all of the JS options to the BREAD, so users can also define the ```separator``` and ```map``` without editing the view. I can work on that later.

File: src/Helpers/helperSlugify.php (hunk 3 above)
Comment (akazorg) [reply src/Helpers/helperSlugify.php:37]: ok  @marktopper

**Comment by marktopper** [comment]
We can always do that later as another pull request.

Merging this in, good job 👍 

__EDIT__: Will do the documentation later

**Pull request merged**